apps/wallet/src/ui/app/hooks/useAccountGroups.ts

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useMemo } from 'react';
+import { useAccounts } from './useAccounts';
+import { defaultSortOrder, groupByType } from '../helpers/sortAccounts';
+
+export function useAccountGroups() {
+	const { data: accounts } = useAccounts();
+
+	const sortedAndGroupedAccounts = useMemo(() => {
+		return groupByType(accounts ?? []);
+	}, [accounts]);
+
+	const list = () => {
```

Comment:
We could memo this and just return the result instead of the function that generates the list. To avoid creating a new array every time